index.js

```diff
@@ -0,0 +1,92 @@
+const url = require('url');
+
+/**
+ * Gate requests based on CORS data. For requests that are not permitted via CORS, invoke the
+ * failure options callback, which defaults to rejecting the request.
+ *
+ * @param {Object} options
+ * @param {String} options.origin The origin of the server - requests from this origin will always
+ *   proceed.
+ * @param {Boolean=} options.strict Whether to reject requests that lack an Origin header. Defaults
+ *   to true.
+ * @param {Boolean=} options.allowSafe Whether to enforce the strict mode for safe requests (HEAD,
+ *   GET). Defaults to true.
+ * @param {Function=} options.failure A standard connect-style callback for handling failure.
```

Comment:
The docs here are great but can you document this in the README too? So someone doesn't have to go search the code.